php/libraries/SinglePointLogin.class.inc

```diff
@@ -308,86 +308,95 @@ class SinglePointLogin
+            return false;
 
-                // check if the account is active
-                if (($row['active_from'] != null)
-                    && ($currentDay < strtotime($row['active_from']))
-                ) {
-                    $this->_lastError = "Your account is not active yet."
-                        ." According to our records it will be active from 
-			{$row['active_from']}"
-                        . " Please contact your project administrator";
-                    return false;
+        }
 
-                }
+        if ($row['Pending_approval'] == 'Y') {
+            $this->_lastError = "Your account has not yet been activated."
+                . " Please contact your project administrator to activate"
+                . " this account.";
+            $this->insertFailedDetail("user account pending", $setArray);
+            return false;
+        }
+        if ($row['Active'] == 'N'
+            || $this->disabledDueToInactivity($username, 365)
+        ) {
+            $this->_lastError = "Your account has been deactivated."
+                . " Please contact your project administrator to"
+                . " reactivate this account.";
+            $this->insertFailedDetail(
+                "user account not active",
+                $setArray
+            );
 
-                if ($row['Pending_approval'] == 'Y') {
-                    $this->_lastError = "Your account has not yet been activated."
-                        . " Please contact your project administrator to activate"
-                        . " this account.";
-                    $this->insertFailedDetail("user account pending", $setArray);
-                    return false;
-                }
+            return false;
+        }
 
-                // save the new password if the last password expired
-                if ($row['Expired']) {
-                    $success = $this->save();
+        // save the new password if the last password expired
+        if ($row['Expired']) {
+            $success = $this->save();
 
-                    if ($success == false) {
-                        if ($redirect) {
-                            $this->showPasswordExpiryScreen();
-                        }
-                        return false;
-                    }
+            if ($success == false) {
+                if ($redirect) {
+                    $this->showPasswordExpiryScreen();
                 }
-
-                // user is valid
-                $this->_username = $row['UserID'];
-
-                $DB->insert('user_login_history', $setArray);
-                return true;
+                return false;
             }
         }
 
-        // bad usename or password
-        $this->_lastError = "Incorrect email or password";
-        return false;
+        // user is valid
+        $this->_username = $row['UserID'];
+
+        $DB->insert('user_login_history', $setArray);
+        return true;
```

Comment:
This refactoring essentially makes the authentication default to "true" unless a blacklisted set of conditions is met rather than defaulting to "false" unless a whitelisted set of conditions are met. This seems dangerous.

Reply:
That's a good point. I changed my approach to fail by default and structure the logic more similarly to how it was previously.